### PR TITLE
less aggessive spam prevention

### PR DESCRIPTION
**Pull request by sqweelygig**
* if an originating comment includes the robot's name, the response shouldn't be spam filtered

### CODE REVIEW

**Comment by sqweelygig** [comment]
Marshmallows!  Would it be possible for this to receive some TLC today; I've got an ASAP change from Alex to go out and I'd very much like to bundle it with these other changes and then leave Hubot alone for a few weeks.

File: src/repeat-prevent.coffee [hunk 1]
```diff
@@ -21,8 +21,8 @@ module.exports = (robot) ->
 		# Tidy the old comments
 		for comment, timestamp of scopeConsidered when timestamp.isBefore(horizon)
 			delete scopeConsidered[comment]
-		# Allow the response if it's not in our memory
-		if (not scopeConsidered[response]?)
+		# Allow the response if bot name is mentioned or it's not in our memory
+		if (context.response.message.text.toLowerCase().includes(context.response.robot.name.toLowerCase())) or (not scopeConsidered[response]?)
```
Comment (pimterry) [inline src/repeat-prevent.coffee]: Nit: Robot name here should probably come out as a separate var, to make this a little more readable.

**Review by pimterry — COMMENTED** [review]
Why is this the case? It's not clear to me why messages that include the bot's name don't count as a duplicated responses.

More generally, we should make sure that the motivating issues on these repos (like https://github.com/resin-io/hubot-as-mainbot/issues/113 here) include their underlying motivation, not just a description of the target implementation. It'd be great to have a specific example of when this is important behaviour, preferably with an illustrative flowdock thread, both for now and as useful context in future.

**Comment by sqweelygig** [comment]
Example behaviours.
The key is looking at the originating comment, and if that specifically mentions the bot, then let whatever the response is through.
Before this change:
```
me> hubot echo test
it> test
me> hubot echo test
me> @test
it> @test is on leave today
me> @test
```
After this change:
```
me> hubot echo test
it> test
me> hubot echo test
it> test
me> @test
it> @test is on leave today
me> @test
```

**Comment by sqweelygig** [comment]
@pimterry

**Comment by CameronDiver** [comment]
@sqweelygig Looking at that example you've posted, it seems that the only difference is that hubot repeated the `test` part twice, which doesn't seem to match what you've written. Would you mind clarifying slightly what the difference is?

**Comment by sqweelygig** [comment]
In the first code block all of the repeated proto-responses are spam filtered.  There would be a second `it> test` and `it> @test is on leave today`, but they get blocked.  This causes a problem when some scripts, eg the London lock, have generic sections to their response.  The way I see it is that if the comment that triggered hubot into action contains the name of the hubot then none of the triggered responses should be filtered.  This is what we see in the second codeblock, where because the echo text contains `hubot` the response is not spam filtered.
Perhaps the following interactions make this more explicit:
```
me> hubot echo test
it (allowed)> test
me> hubot echo test
it (filtered)> test
me> @test
it (allowed)> @test is on leave today
me> @test
it (filtered)> @test is on leave today
```
Note that Line 4 becomes allowed through the filter because Line 3 contains `hubot`
```
me> hubot echo test
it (allowed)> test
me> hubot echo test
it (allowed)> test
me> @test
it (allowed)> @test is on leave today
me> @test
it (filtered)> @test is on leave today
```

**Review by CameronDiver — APPROVED** [review]
LGTM

**Comment by lurch** [comment]
Are there ever any scenarios (in which you want the output to not be filtered) in which the `botName` (either `hubot` or `@hubot`) _isn't_ the first thing in the `originalComment` ?

**Pull request merged**